crates/subspace-farmer/src/plotter/cpu.rs

```diff
@@ -456,12 +457,19 @@ where
                         SectorPlottingProgress::Finished {
                             plotted_sector,
                             time: start.elapsed(),
-                            sector: Box::pin(stream::once(async move { Ok(Bytes::from(sector)) })),
+                            sector: Box::pin({
+                                let mut sector = Some(Ok(Bytes::from(sector)));
+
+                                stream::poll_fn(move |_cx| {
+                                    // Just so that permit is dropped with stream itself
+                                    let _downloading_permit = &downloading_permit;
```

Comment:
This will not work because this is `FnMut` callback, it will be called more than once. I specifically wanted to simply bind the lifetime of the permit to the lifetime of the stream and this is the simplest way I could come up with :sweat_smile: